server/business-process-query/get-data.js

```diff
@@ -125,24 +156,36 @@ module.exports = exports = memoize(function (driver) {
 				} else {
 					keyPath = id.slice(bpId.length + 1);
 				}
+
+				// Business process properties
 				meta = exports.businessProcessMetaMap[keyPath];
-				if (meta) {
-					if (meta.type && (meta.type !== 'direct')) return;
-					if (multiItemValue && !meta.multiple) return;
-					if (!meta.validate(record, multiItemValue)) return;
+				if (validateRecord(record, meta, multiItemValue)) {
 					meta.set(initBpDataset(bpId), record, multiItemValue);
+					return;
+				}
+				// Certificates
+				match = keyPath.match(certificatePropertyRe);
+				if (match) {
+					certificatePath = match[1];
+
+					initCertDataset(certificatePath, bpId);
+				}
+
+				// Processing steps
+				match = keyPath.match(processingStepPropertyRe);
+				if (match) {
+					stepPath = match[1];
+					stepKeyPath = match[2];
+
+					if (!stepPaths.has(stepPath)) return;
+
+					// Processing step properties
+					meta = exports.stepMetaMap[stepKeyPath];
+					if (validateRecord(record, meta, multiItemValue)) {
+						meta.set(initStepDataset(stepPath, bpId), record, multiItemValue);
+						return;
+					}
```

Comment:
> To properly compute the dateMap I need all statusHistory logs for each BP that had changes.

Yes, I  explained in https://github.com/egovernment/eregistrations/pull/1758#issuecomment-283069864 how you can query for this data specifically without doing whole db scan, it's straightforward.

We need to maintain one db scan done at system initialization, and do not introduce any further scans over process run (`business-process-query` was introduced purely because [we had performance issues caused by many db scans](https://github.com/egovernment/eregistrations/pull/1602))